src/contracts/orderbook/NiftyswapOrderbook.sol

```diff
@@ -0,0 +1,422 @@
+        address[] memory additionalFeeRecievers
+    ) external {
+        Order memory listing = orders[listingId];
+        if (!listing.isListing || listing.creator == address(0)) {
+            // Is a listing, cancelled, completed or never existed
+            revert InvalidListingId(listingId);
+        }
+
+        _acceptOrder(listingId, listing, quantity, additionalFees, additionalFeeRecievers);
+
+        emit ListingAccepted(listingId, msg.sender, listing.tokenContract, quantity);
+    }
+
+    /**
+     * Sells a token.
+     * @param offerId The ID of the listing.
+     * @param quantity The quantity of tokens to sell.
+     * @param additionalFees The additional fees to pay.
+     * @param additionalFeeRecievers The addresses to send the additional fees to.
+     */
+    function acceptOffer(
+        bytes32 offerId,
+        uint256 quantity,
+        uint256[] memory additionalFees,
+        address[] memory additionalFeeRecievers
+    ) external {
+        Order memory offer = orders[offerId];
+        if (offer.isListing || offer.creator == address(0)) {
+            // Is a listing, cancelled, completed or never existed
+            revert InvalidOfferId(offerId);
+        }
+
+        _acceptOrder(offerId, offer, quantity, additionalFees, additionalFeeRecievers);
+
+        emit OfferAccepted(offerId, msg.sender, offer.tokenContract, quantity);
+    }
+
+    function _acceptOrder(
+        bytes32 orderId,
+        Order memory order,
+        uint256 quantity,
+        uint256[] memory additionalFees,
+        address[] memory additionalFeeRecievers
+    ) internal {
+        if (quantity == 0 || quantity > order.quantity) {
+            revert InvalidQuantity();
+        }
+        if (_isExpired(order)) {
+            revert InvalidExpiry();
+        }
+        if (additionalFees.length != additionalFeeRecievers.length) {
+            revert InvalidAdditionalFees();
+        }
+
+        // Update order state
+        if (order.quantity == quantity) {
+            // Refund some gas
+            delete orders[orderId];
+        } else {
+            orders[orderId].quantity -= quantity;
+        }
+
+        // Calculate payables
+        uint256 remainingCost = order.pricePerToken * quantity;
+        (address royaltyRecipient, uint256 royaltyAmount) =
+            getRoyaltyInfo(order.tokenContract, order.tokenId, remainingCost);
+
+        address currencyReceiver = order.isListing ? order.creator : msg.sender;
+        address tokenReceiver = order.isListing ? msg.sender : order.creator;
+
+        if (royaltyAmount > 0) {
+            // Transfer royalties
+            TransferHelper.safeTransferFrom(order.currency, tokenReceiver, royaltyRecipient, royaltyAmount);
+            remainingCost -= royaltyAmount;
+        }
+
+        // Transfer additional fees
+        uint256 feesPaid = 0;
+        for (uint256 i; i < additionalFees.length; i++) {
+            uint256 fee = additionalFees[i];
```

Comment:
Why `fee` as a variable but not `additionalReceiver` ?

Reply:
And I spelt receiver wrong 🤦